frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/auto-detect-volume.tsx

```diff
@@ -1,22 +1,44 @@
 import * as React from 'react';
+import * as _ from 'lodash';
 import { Form } from '@patternfly/react-core';
+import { NodeKind } from '@console/internal/module/k8s';
+import { getName } from '@console/shared';
 import {
   AutoDetectVolumeInner,
   AutoDetectVolumeHeader,
 } from '@console/local-storage-operator-plugin/src/components/auto-detect-volume/auto-detect-volume-inner';
-import { State, Action } from '../state';
+import { hasOCSTaint } from '../../../../../utils/install';
 import { RequestErrors } from '../../../install-wizard/review-and-create';
+import { State, Action } from '../state';
 import '../../attached-devices.scss';
 
-export const AutoDetectVolume: React.FC<AutoDetectVolumeProps> = ({ state, dispatch }) => (
-  <>
-    <AutoDetectVolumeHeader />
-    <Form noValidate={false} className="ceph-ocs-install__auto-detect-table">
-      <AutoDetectVolumeInner state={state} dispatch={dispatch} />
-    </Form>
-    <RequestErrors errorMessage={state.error} inProgress={state.isLoading} />
-  </>
-);
+export const AutoDetectVolume: React.FC<AutoDetectVolumeProps> = ({ state, dispatch }) => {
+  const setOCSTaint = (nodes: NodeKind[]): void => {
+    const newTaintedNodes = nodes.reduce((taintedNodes, node) => {
+      if (hasOCSTaint(node)) taintedNodes.push(getName(node));
+      return taintedNodes;
+    }, []);
+    if (!_.isEqual(state.taintedNodesForADV, newTaintedNodes)) {
+      dispatch({ type: 'setTaintedNodesForADV', value: newTaintedNodes });
+    }
+  };
```

Comment:
why do we need to set the setTaintedNodesForADV? We just can set the tolaration in ADV or LVS if it's created from OCS storage cluster creation.

Reply:
Not really, if a tainted node is chosen for discovery, then we cant predict that user will choose it for creating local volume set too

Reply:
Also if the discovery is triggered from OCS we can assume and add the tolerations. Also, there is no disadvantage to adding toleration if done.  IMO

Reply:
I am not sure of disadvantages , but surely it would end up setting a configuration of yaml thats not asked for.
A bit skeptical about this approach.